devlib/instrument/__init__.py

```diff
@@ -72,9 +72,25 @@ def __repr__(self):
             return text.format(self.name, self.units)
 
 
-# Standard measures
+# Standard measures. In order to make sure that downstream data processing is not tied
+# to particular insturments (e.g. a particular method of mearuing power), instruments
+# must, where possible, resport their measurments formatted as on of the standard types
+# defined here.
 _measurement_types = [
+    # For whatever reason, the type of measurement could not be established.
     MeasurementType('unknown', None),
+
+    # Generic measurements
+    MeasurementType('count', 'count'),
+    MeasurementType('percent', 'percent'),
+
+    # Time measurement. While there is typically a single "canonical" unit
+    # used for each type of measurmenent, time may be measured to a wide variety
+    # of events occuring at a wide range of scales. Forcing everying into a
+    # single scale will lead to inefficient and awkward to work with result tables.
+    # Coversion functions between the formats are specified, so that downstream
+    # processors that expect all times time be at a particular scale can automatically
+    # covert without being familar with individual instruments.
```

Comment:
I wonder if it would be useful to differentiate between times that are periods ('this frame took 12ms to render') and times that are stamps ('this frame was rendered at t=1005.3s')?

Reply:
hm, possibly, however, periods can also be large ("this gem5 simulation took several days to run"), so it would not be enough to simply have "time" measurement in "seconds" and "period" measurement in "microseconds". You'd arguably still need the three levels here.

It may be useful to add an additional "timestamp" measurement type, but again, I'm hesitant to do so without something that actually requires it. In order for that to make sense if there is something that cares about absolute vs relative time difference, but not about the other details of the metric (otherwise just use metric name).

Reply:
OK, that makes sense too.